beetsplug/zero.py

```diff
@@ -123,9 +124,15 @@ def set_fields(self, item, tags):
         """
         fields_set = False
 
+        if "disc" in tags and self.config["omit_single_disc"].get(bool):
+            if item.disctotal == 1:
+                fields_set = True
+                self._log.debug("disc: {.disc} -> None", item)
+                tags["disc"] = None
+
         if not self.fields_to_progs:
-            self._log.warning("no fields, nothing to do")
-            return False
+            self._log.warning("no fields list to remove")
+            return fields_set
```

Comment:
I think we can remove the return here. Should make thinking about the code flow easier. If `fields_to_progs` is empty the iteration below is not done anyways. We should keep the log message tho.